src/app/credExplorer/pagerankTable/Aggregation.js

```diff
@@ -0,0 +1,135 @@
+    );
+    function aggregationKey(aggregation: FlatAggregation) {
+      const result: any = {
+        nodePrefix: aggregation.nodeType.prefix,
+      };
+      switch (aggregation.connectionType.type) {
+        case "SYNTHETIC_LOOP":
+          result.connectionType = {type: "SYNTHETIC_LOOP"};
+          break;
+        case "IN_EDGE":
+          result.connectionType = {
+            type: "IN_EDGE",
+            edgePrefix: aggregation.connectionType.edgeType.prefix,
+          };
+          break;
+        case "OUT_EDGE":
+          result.connectionType = {
+            type: "OUT_EDGE",
+            edgePrefix: aggregation.connectionType.edgeType.prefix,
+          };
+          break;
+        default:
+          throw new Error((aggregation.connectionType.type: empty));
+      }
+      return JSON.stringify(result);
+    }
+    return (
+      <React.Fragment>
+        {aggregations.map((agg) => (
+          <AggregationRow
+            key={aggregationKey(agg)}
+            depth={depth}
+            target={node}
+            sharedProps={sharedProps}
+            aggregation={agg}
+          />
+        ))}
+      </React.Fragment>
+    );
+  }
+}
+
+type AggregationRowProps = {|
+  +depth: number,
+  +target: NodeAddressT,
+  +aggregation: FlatAggregation,
+  +sharedProps: SharedProps,
+|};
+
+export class AggregationRow extends React.PureComponent<AggregationRowProps> {
+  render() {
+    const {sharedProps, target, depth, aggregation} = this.props;
+    const {pnd} = sharedProps;
+    const score = aggregation.summary.score;
+    const {score: targetScore} = NullUtil.get(pnd.get(target));
+    const connectionProportion = score / targetScore;
+
+    return (
+      <TableRow
+        depth={depth}
+        indent={1}
+        showPadding={false}
+        connectionProportion={connectionProportion}
+        score={score}
+        description={<AggregationView aggregation={aggregation} />}
+      >
+        <ConnectionRowList
+          key="children"
+          depth={depth}
+          node={target}
+          connections={aggregation.connections}
+          sharedProps={sharedProps}
+        />
+      </TableRow>
+    );
+  }
+}
+
+export class AggregationView extends React.PureComponent<{|
+  +aggregation: FlatAggregation,
+|}> {
+  render() {
+    const {aggregation} = this.props;
+    const {connectionType, summary, nodeType} = aggregation;
+    function connectionDescription() {
+      switch (connectionType.type) {
+        case "SYNTHETIC_LOOP":
+          return "synthetic loop";
+        case "IN_EDGE":
+          return connectionType.edgeType.backwardName;
+        case "OUT_EDGE":
+          return connectionType.edgeType.forwardName;
+        default:
+          throw new Error((connectionType.type: empty));
+      }
+    }
+    return (
+      <span>
+        <Badge>{connectionDescription()}</Badge>
+        <span> {summary.size} </span>
+        <span style={{textTransform: "lowercase"}}>{nodeType.name}</span>
```

Comment:
This transformation is highly questionable in general, and we will
probably want to remove it later, but it is an improvement over not
having it right now, so I’m okay with including it.

Reply:
Ack